docs/DeveloperGuide.md

```diff
@@ -1,38 +1,180 @@
+
+Step 2. The user executes `expense /amt 20 /date 24-10-2023 /desc Lunch /note Pasta /cat Food` command to create a
+transaction. The `expense` command calls `Parser#parseExpense()` to create an `Expense` object. The 
+`AddCommand#execute()` is then called to store the `Expense` object in the `TransactionList`.
+
+The following sequence diagram shows how the add transaction operation works:
+
+<img src="images/AddTransactionSequenceDiagram.png" width="600" />
+
+### View transaction feature
+
+#### I. Architecture-Level Design
+The "View Transaction" feature primarily involves the following components:
+
+1. Parser: Responsible for interpreting user input and generating a ViewCommand object.
+
+2. ViewCommand: A subclass of the Command class, created by the Parser to represent the "view" command.
+
+3. Nuscents: The main application class that receives and executes commands. It invokes the execute() method of the ViewCommand.
+
+4. TransactionList: A data structure to store and manage transactions.
+
+5. Transaction: Represents individual transactions.
+
+6. UI: Handles user interface and messaging.
+
+#### II. Component-Level Design
+1. Parser
+
+The Parser class identifies the "view" command and extracts the taskIndex (transaction index) from the user's input.
+
+2. ViewCommand
+
+The ViewCommand object is created by the Parser. It encapsulates the user's request to view a specific transaction. This object is passed to the Nuscents class for execution.
+
+3. Nuscents
+
+In the Nuscents class, the execute() method of the ViewCommand is called, and the taskIndex is extracted from the command. It then calls the viewTransaction(taskIndex) method on the TransactionList.
+
+4. TransactionList
+
+The TransactionList contains a list of Transaction objects. The viewTransaction(taskIndex) method retrieves the specific Transaction object based on the taskIndex.
+
+5. Transaction
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The Transaction class represents an individual transaction, and it contains all the relevant details of a transaction.
+
+6. UI
+
+The UI class displays the transaction details using the UI.showTransactionViewMessage method, which receives the Transaction object as input and presents the detailed transaction information to the user.
+
+#### III. Alternatives Considered
+While the current design is deemed suitable for our application, we did consider alternative approaches, such as 
+integrating the view transaction functionality directly within the Nuscents class without introducing a ViewCommand. 
+However, we opted for the current design to promote a cleaner separation of concerns and to facilitate future expansions 
+and modifications.
+
+The following sequence diagram shows how the add transaction operation works:
+<img src="images/ViewSequenceDiagram.png" width="600" />
```

Comment:
The sentence says "how the add transaction operation works" but the sequence diagram is about the ViewCommand. Maybe that's an error?

<img width="496" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/9a7f2330-dd60-4b1e-8937-b0e1df39f133">